lib/prometheus-parser/src/lib.rs

```diff
@@ -79,7 +79,7 @@ pub struct SummaryMetric {
     pub count: u32,
 }
 
-#[derive(Debug, Default, PartialEq)]
+#[derive(Debug, Default, PartialEq, PartialOrd)]
```

Comment:
Is there a reason we don't just derive or implement `Ord`?

Reply:
The "total order" predicate required for `Ord` semantics on floating point values is not implemented in hardware, so actually implementing this in Rust is messy at best. See https://stackoverflow.com/questions/26489701/why-does-rust-not-implement-total-ordering-via-the-ord-trait-for-f64-and-f32

Reply:
FTR I created this based on the suggestion in the above link, which passes the trivial test, but I'm not confident it's actually correct for negatives and such. What's in this patch is much simpler.
```rust
impl Eq for HistogramBucket {}

impl Ord for HistogramBucket {
    fn cmp(&self, other: &Self) -> Ordering {
        self.bucket
            .is_sign_positive()
            .cmp(&other.bucket.is_sign_positive())
            .then_with(|| {
                (self.bucket.to_bits() | 0x7fffffffffffffffu64)
                    .cmp(&(other.bucket.to_bits() & 0x7fffffffffffffffu64))
            })
            .then_with(|| self.count.cmp(&other.count))
    }
}
```

Reply:
Aha, yes, I see. I neglected the bucket was an `f64`. In that case, I think what you have is good.